File: vite/minify.py
import requests

filename = "ShogiCross.mjs"
filename_min = "ShogiCross.min.mjs"
with open(filename, "r") as r:
	response = requests.post(
		'https://www.toptal.com/developers/javascript-minifier/api/raw',
		data=dict(input=r.read())
	).text

with open(filename_min, "w") as w:
	w.write(response)

File: vite/minify.js
import path from "path";
import fs from "fs/promises";
import querystring from "querystring";

const inputName = process.argv[2];
const baseName = `${path.dirname(inputName)}/${path.parse(inputName).name}`;
await fs.rename(inputName, `${baseName}.js`)

const code = await fs.readFile(`${baseName}.js`, {encoding: "utf8"});
const response = await fetch(
	"https://www.toptal.com/developers/javascript-minifier/api/raw",
	{
		method: "POST",
		headers: {
			'Content-Type': 'application/x-www-form-urlencoded'
		},
		body: querystring.stringify({
			input: code
		})
	}
);
const minifyCode = await response.text();
await fs.writeFile(`${baseName}.min.js`, minifyCode);
console.log("Minify Success!")